new-lamassu-admin/src/pages/Maintenance/MachineDetailsCard.js

```diff
@@ -69,22 +97,63 @@ const Item = ({ children, ...props }) => (
 )
 
 const MachineDetailsRow = ({ it: machine, onActionSuccess }) => {
-  const [action, setAction] = useState(null)
+  const [action, setAction] = useState({ command: null })
   const [errorMessage, setErrorMessage] = useState(null)
   const classes = useMDStyles()
 
+  const [
+    fetchMachineEvents,
+    { loading: loadingEvents, data: machineEventsLazy }
+  ] = useLazyQuery(MACHINE, {
+    variables: {
+      deviceId: machine.deviceId
+    }
+  })
+
+  useEffect(() => {
+    if (action.command === 'restartServices') {
+      fetchMachineEvents()
+    }
+  }, [action.command, fetchMachineEvents])
```

Comment:
since fetchMachineEvents is a common local variable we don't really need a use effect here.